api/src/main/java/org/msf/module/visit/CloseVisitOnAnOutcomeTask.java

```diff
@@ -49,8 +57,87 @@ public void execute() {
 
     }
 
+    private boolean isProgramStateConfiguredAndHasHospitalVisit() {
+        return !visitCloseData.getProgramStateConcepts().isEmpty() && containsHospitalVisitType();
+    }
+
+    private boolean containsHospitalVisitType() {
+        return visitCloseData.getVisitTypes().stream().anyMatch(visitType ->
+                visitType.getName().equals(HOSPITAL_VISIT_TYPE_AMMAN));
+    }
+
+    private void closeVisitForAmman(List<Visit> visits) {
+        for (Visit openVisit : visits) {
+            ProgramWorkflowService programWorkflowService = Context.getService(ProgramWorkflowService.class);
+            List<PatientProgram> activePatientProgramList = getActivePatientProgramForPatient(openVisit.getPatient(), programWorkflowService);
+            if (!activePatientProgramList.isEmpty()) {
+                if (isHospitalVisit(openVisit)) {
+                    if (isNotInAnyConfiguredProgramStateOrBedIsAssigned(programWorkflowService, activePatientProgramList, openVisit.getPatient())) {
+                        continue;
+                    }
+                } else {
+                    if (!outcomeAvailable(openVisit, visitCloseData.getOutcomeConcepts()))
+                        continue;
+                }
+            }
+            visitService.endVisit(openVisit, new Date());
+        }
+    }
+
+    private boolean isNotInAnyConfiguredProgramStateOrBedIsAssigned(ProgramWorkflowService programWorkflowService, List<PatientProgram> activePatientProgramList, Patient patient) {
+        if (activePatientProgramList.size() > 1) {
```

Comment:
@binduak @vinishad Can we have null check for this condition something like "if (activePatientProgramList && activePatientProgramList.size() > 1)"

Reply:
@tarunkumar-tw @supriyatw If the list is empty, it will not even come here. pls have a look at line number 78.